### ISSUE TITLE
🧐Multi line Chart map data source to array objects with KEY-VALUE pairs

### ISSUE THREAD

**Issue by leventebg93** [ant-design/ant-design-charts#1565]
### 🧐 问题描述 [详细地描述问题，让大家都能理解]
Currently this is how the data mapping works for [this chart ](https://charts.ant.design/en/examples/line/multiple#marker-active)
we have data like this:
```
[{
"year": "1850",
"value": 0,
"category": "Liquid fuel"
},
{
"year": "1850",
"value": 54,
"category": "Solid fuel"
},
{
"year": "1850",
"value": 0,
"category": "Gas fuel"
}] ....,
```
That'll give us this chart
![imgonline-com-ua-resize-yXfjy2oV30SZn](https://user-images.githubusercontent.com/97885059/189917145-ccef737a-eb80-4dc8-a69b-dfd6c9fa3819.jpg)

Is it possible for the data source to be formatted like this INSTEAD:
```
{
"year": "1850",
"Liquid fuel": 0,
"Solid fuel": 54,
"Gas fuel":0
},
{
"year": "1851",
"Liquid fuel": 3,
"Solid fuel": 60,
"Gas fuel":3
}...
,
```
So in short multiple items of the same year are combined into one objects with KEY VALUE pairs where the category is used as "KEY" and the value field used as "VALUE" 

Is it possible to map this array of objects to the chart?

**Comment by lxfu1** [comment]
The upper layer processes the data, eg:
```ts
  const dataSet = (d: object[], categroy: string) => {
    const res = [];
    d.forEach(item => {
      const { year, ...rest } = item;
      const pairArr = Object.keys(rest).map(k => {
        return {
          year: item.year,
          value: item[k],
          [categroy]: k
        }
      })
      res.push(...pairArr)
    })
    return res;
  }
  console.log(dataSet(data, 'type'));
```

**Comment by leventebg93** [comment]
> {
> "year": "1850",
> "Liquid fuel": 0,
> "Solid fuel": 54,
> "Gas fuel":0
> },
> {
> "year": "1851",
> "Liquid fuel": 3,
> "Solid fuel": 60,
> "Gas fuel":3
> }

I see what you are doing there, I thought there was an simpler built-in solution to this, but if not then this will help
Thank you!
